src/sagas/nanoContract.js

```diff
@@ -111,9 +111,9 @@ export function* registerNanoContract({ payload }) {
     return;
   }
 
-  let ncState = null;
+  let ncHistory = null;
   try {
-    ncState = yield call([ncApi, ncApi.getNanoContractState], ncId);
+    ncHistory = yield call([ncApi, ncApi.getNanoContractHistory], ncId, 1);
```

Comment:
I understand that you are using this to check if the nano contract exists but semantically I prefer doing this like the desktop wallet.

```
  try {
    const response = yield call([wallet, wallet.getFullTxById], ncId);
    tx = response.tx;
  } catch (error) {
    console.error('Error while registering Nano Contract.', error);
    yield put(nanoContractRegisterError(t`Error while registering the nano contract.`));
    return;
  }

  if (!nanoUtils.isNanoContractCreate(tx)) {
    console.debug('Transaction is not a nano contract creation transaction.');
    yield put(nanoContractRegisterError(t`Invalid nano contract creation transaction.`));
    return;
  }
```
https://github.com/HathorNetwork/hathor-wallet/blob/master/src/sagas/nanoContract.js

What do you think?

Reply:
Replaced: 4b81d31958257d4050a6af348bf85023fc5ed77c